pkg/controlplane/apiserver/apis.go

```diff
@@ -83,8 +90,55 @@ func (c *CompletedConfig) GenericStorageProviders(discovery discovery.DiscoveryI
 	}, nil
 }
 
+// StorageReadinessHook implements PostStartHook functionality for checking readiness
+// of underlying storage for registered resources.
+//
+// FIXME: Figure out where to put it.
+type StorageReadinessHook struct {
+	checks  map[string]func() error
+	timeout time.Duration
+}
+
+// NewStorageReadinessHook created new StorageReadinessHook.
+func NewStorageReadinessHook(timeout time.Duration) *StorageReadinessHook {
+	return &StorageReadinessHook{
+		checks:  make(map[string]func() error),
+		timeout: timeout,
+	}
+}
+
+func (h *StorageReadinessHook) RegisterStorage(gvr metav1.GroupVersionResource, storage rest.StorageWithReadiness) {
+	// FIXME: handle duplicates
+	h.checks[gvr.String()] = storage.ReadinessCheck
```

Comment:
For duplicates, we're not expecting there to be duplicates when everything is configured correctly?  We just need to handle the unexpected case?